### ISSUE TITLE
Electrum 2.5.1 forgets the imported addresses after restart on Linux Machines

### ISSUE THREAD

**Issue by twisterswirl** [spesmilo/electrum#1511]
I was having similar problem with the 2.4.4 version also but updated to see if the newer one resolves it but it doesn't.

The issue is, if I import a private key into electrum running on linux machine, the address remains available to use for the time the electrum stays online, after the electrum is restarted it forgets the address and there is no trace of it in the wallet.

Kindly fix this issue as it is really annoying to re add keys again and again. Thanks

**Comment by ecdsa** [comment]
I cannot reproduce that issue.
Did you import the key into a wallet already containing imported keys, or in an old wallet with seed?

**Comment by ecdsa** [comment]
also, do you have permissions to write the wallet file?

**Comment by ecdsa** [comment]
closing: no answer from OP, and I think it's a permission problem.

**Comment by twisterswirl** [comment]
Sorry, I was not able to respond earlier.

-- Yes, the wallet have other keys which were imported earlier, I am using the same wallet file not restoring it with the seed.

-- Yes I am root.

**Comment by ecdsa** [comment]
ok, can you run electrum from a terminal and check if there is any traceback in your session?

**Comment by twisterswirl** [comment]
I'll try and see.

**Comment by twisterswirl** [comment]
Ok,  I think I found the problem, when I run it just by clicking the icon from the menu, it runs as not root and the address does not gets added (or maybe it gets added temporarily and not saved because it doesn't have the permission to save) to the wallet file. When I ran it from the root in terminal, it wrote the address and it's encrypted key in the wallet file. So, I think you were right, the problem was the permission.

This is solved.
